src/bun.js/webcore/response.zig

```diff
@@ -708,9 +720,100 @@ pub const Fetch = struct {
             this.javascript_vm.allocator.destroy(this);
         }
 
-        pub fn onDone(this: *FetchTasklet) void {
+        pub fn onBodyReceived(this: *FetchTasklet) void {
+            if (this.aborted.load(.Acquire)) return;
+
+            const success = this.result.isSuccess();
+            const globalThis = this.global_this;
+            defer {
+                if (!success or !this.result.has_more) {
+                    var vm = globalThis.bunVM();
+                    this.poll_ref.unref(vm);
+                    this.clearData();
+                    this.deinit();
+                }
+            }
+
+            if (!success) {
+                const err = this.onReject();
+                globalThis.throwValue(err);
+                return;
+            }
+
+            if (this.response) |response| {
+                const body = response.body;
+                if (body.value == .Locked) {
+                    if (body.value.Locked.readable) |readable| {
+                        if (readable.ptr == .Bytes) {
+                            this.response.?.body.value.Locked.readable.?.ptr.Bytes.size_hint = @as(u52, @intCast(this.body_size));
+                            var scheduled_response_buffer = this.scheduled_response_buffer.list;
+
+                            const chunk = scheduled_response_buffer.items;
+
+                            if (this.result.has_more) {
+                                readable.ptr.Bytes.onData(
+                                    .{
+                                        .temporary = bun.ByteList.initConst(chunk),
+                                    },
+                                    bun.default_allocator,
+                                );
+
+                                // clean for reuse later
+                                this.scheduled_response_buffer.reset();
+                            } else {
+                                readable.ptr.Bytes.onData(
+                                    .{
+                                        .temporary_and_done = bun.ByteList.initConst(chunk),
+                                    },
+                                    bun.default_allocator,
+                                );
+                            }
+
+                            return;
+                        }
+                    } else {
+                        this.response.?.body.value.Locked.size_hint = @as(u52, @intCast(this.body_size));
```

Comment:
is the pointer aliasing here intentional?

Reply:
fixed https://github.com/oven-sh/bun/pull/4127/commits/4676cfe8c877ca99e2fb4f2598afb51b51ceaadc